lib/gollum-lib/filter/critic_markup.rb

```diff
@@ -0,0 +1,93 @@
+# ~*~ encoding: utf-8 ~*~
+
+# CriticMarkup
+#
+# Render CriticMarkup
+
+class Gollum::Filter::CriticMarkup < Gollum::Filter
+
+  # Patterns inspired by https://github.com/DivineDominion/criticmarkup.tmbundle/blob/master/Syntaxes/criticmarkup.tmLanguage
+  # All patterns use multiline matching (m flag)
+  # Logic inspired by https://github.com/CriticMarkup/CriticMarkup-toolkit/blob/master/CLI/criticParser_CLI.py
+  
+  ADDITION_PATTERN      = %r|{\+\+(?<content>.*?)\+\+[ \t]*(\[(.*?)\])?[ \t]*\}|m
+  DELETION_PATTERN      = %r|{--(?<content>.*?)--[ \t]*(\[(.*?)\])?[ \t]*\}|m
+  SUBSTITUTION_PATTERN  = %r|{~~(?<oldcontent>.*?)~>(?<newcontent>.*?)~~}|m
+  HIGHLIGHT_PATTERN     = %r|{\=\=(?<content>.*?)[ \t]*(\[(.*?)\])?[ \t]*\=\=\}{>>(?<comment>.*?)<<}|m
+  COMMENT_PATTERN       = %r|{>>(?<content>.*?)<<}|m
+
+  PROCESS_PATTERN       = /(?<placeholder>CRITIC\h+)/
+
+
+  def extract(data)
+    data.gsub! ADDITION_PATTERN do
+      content = $~[:content]
+      placeholder = "CRITIC" + Digest::SHA1.hexdigest("#{content}#{@map.size+1}")
+    	# Is there a new paragraph followed by new text
+      if content.start_with?("\n\n") && content != "\n\n"
```

Comment:
> Is there a need to deal with \r\n? Probably not.

I don't think so.

Reply:
> Another possible edge case: \n \nSome text Or would that be properly treated as new regular text, without a new paragraph? If not, could it be appropriate to use a regexp here, e.g. /\n\s*\n.+/ ?

I see your point, but I'm a little hesitant to deviate from the official spec.

Reply:
> I see your point, but I'm a little hesitant to deviate from the official spec.

👍

Reply:
> Would it help to nest the second conjunct inside the if-statement?

I think nesting does not really help in this case. I suppose it could be a `case/when` but I'm not too keen on that syntax here either.